[VideoInfoScanner][GUIDialogVideoInfo] Update blank movie sets with artwork from Movie Set Information Folder

78andyp opened the pull request:
There are now three ways the movie set artwork can be updated:

1) On a library update - good if you have multiple sets with no artwork.
2) Right click (just the same as choose art)
3) Through the video information dialog (just the same as choose art)

After 3 the library display isn't automatically updated - this is a different bug - see #24176.

## Description
When a movie set is added by the scraper, sometimes there is no associated artwork, for example:

<img width="195" alt="image" src="https://github.com/xbmc/xbmc/assets/99039295/babcdc09-19fb-4e9a-9f7b-f8a0899238ae">

If, when the movie is added, there is an entry in the Movie Set Information Folder (as per https://kodi.wiki/view/Movie_set_information_folder) then any artwork from there is used.

## Motivation and context

Obviously, you are't going to know if there is no set artwork until you've added the movie. Even if you then add artwork in the Movie Set Information Folder - it is never scanned again - you would have to add the artwork manually through the context menu.

If you have a number of sets to update - this is slow.

This PR rescans any movie sets with no atwork, when the library is updated, to see if there is any artwork in the Movie Set Information Folder and adds it automatically.

## How has this been tested?

On my library in Win 11 x64.

## What is the effect on users?

Easier to add movie set artwork - just need to setup the Movie Set Information folder (as per Wiki above) and then place a poster.jpg in a subdiretory with the name of the movie set. It will then be used automatically at next library update.

## Screenshots (if appropriate):

Using above example:

<img width="390" alt="image" src="https://github.com/xbmc/xbmc/assets/99039295/18f301ba-2d26-42e6-a569-5ceeb74cb34b">

After library update:

<img width="219" alt="image" src="https://github.com/xbmc/xbmc/assets/99039295/9903d668-e81d-4bdb-9cb2-eac69fd79deb">

## Types of change
<!--- What type of change does your code introduce? Put an `x` with no space in all the boxes that apply like this: [X] -->
- [ ] **Bug fix** (non-breaking change which fixes an issue)
- [ ] **Clean up** (non-breaking change which removes non-working, unmaintained functionality)
- [ X ] **Improvement** (non-breaking change which improves existing functionality)
- [ ] **New feature** (non-breaking change which adds functionality)
- [ ] **Breaking change** (fix or feature that will cause existing functionality to change)
- [ ] **Cosmetic change** (non-breaking change that doesn't touch code)
- [ ] **Student submission** (PR was done for educational purposes and will be treated as such)
- [ ] **None of the above** (please explain below)

## Checklist:
<!--- Go over all the following points, and put an `X` with no space in all the boxes that apply like this: [X] -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
- [ ] My code follows the **[Code Guidelines](https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md)** of this project 
- [ ] My change requires a change to the documentation, either Doxygen or wiki
- [ ] I have updated the documentation accordingly
- [ ] I have read the **[Contributing](https://github.com/xbmc/xbmc/blob/master/docs/CONTRIBUTING.md)** document
- [ ] I have added tests to cover my change
- [ ] All new and existing tests passed

I'm sure it's probably not in the right place in the code!

ksooo commented:
@78andyp thanks for the PR. Need to say that this code area is not something I'm familiar with, so cannot say anything about correctness of the implementation, especially not, whether the place you added the code is appropriate. I hope somebody else can pick this up for review.

KarellenX commented:
Nice feature. Thanks!!

Is there an option to update individual sets instead of a full library update?

78andyp commented:
> Nice feature. Thanks!!
> 
> Is there an option to update individual sets instead of a full library update?

Do mean like as a context menu option for the individual set?? I can certainly look.

ksooo commented:
More like video info dialog, add support for button refresh.

KarellenX commented:
>Do mean like as a context menu option for the individual set?? 

Actually, like ksooo mentions, the Refresh button. Here is an example i've photoshopped...
My concern would be for users with large libraries having to run an extensive global Library Update to update one or two movie sets.

![setsrefresh](https://github.com/xbmc/xbmc/assets/44626605/18a8b62c-b275-41eb-96b7-0fdcfd7bca59)

78andyp commented:
Thanks both.

I'd actually started playing with a context menu option.

With the next commit, the option 'Use local art' will appear when the set has no art currently AND there is a directory in the MSIF.

<img width="419" alt="image" src="https://github.com/xbmc/xbmc/assets/99039295/3d7c7fa3-01da-4ee8-a130-76d5e3d09a36">

I will look at a refresh button as well.

hurricane51 commented:
So it will only populate BLANK movie sets? I did a clean re-install of KODI 19, but my media all had set information in their NFO files. The sets were thus recreated. but KODI generated the set art. I already have a fully populated MSIF folder. How do I reassign the art in my MSIF folder to the existing sets?

78andyp commented:
> So it will only populate BLANK movie sets? I did a clean re-install of KODI 19, but my media all had set information in their NFO files. The sets were thus recreated. but KODI generated the set art. I already have a fully populated MSIF folder. How do I reassign the art in my MSIF folder to the existing sets?

Can I just check a couple of things:

You have the MSIF set in Settings -> Media -> Videos -> Movie Set Information Folder??
You have Local Information Only set in the Choose information provider option of the Sources??

If both of those are true then it feels like Kodi should look in the MSIF. I'll see what I can recreate here.

hurricane51 commented:
Yes to both of those. I have been having an exchange on the KODI forum, and here’s what I posted:
 
**************************************** 
 
I did a new install of KODI 19. 
 
I configured everything before adding any content, and that included specifying the MSIF folder. The MSIF had been created with my previous installation, and I saw no need to re-create it for the same content.
 
My movies all had NFO files along with the video in their individual folder. These NFO's all had set information. 
 
I wasn't creating any new sets.
 
I initially specified Use Local Information as my information provider to grab the info from the local files. After the movies were in the library, I changed the scraper to Universal Movie scraper for future additions.
 
I found that KODI had generated/assigned random art (from the included movies) to most of the existing sets. Others, it left blank. It didn't use the MSIF folder art.
 
I would like to avoid having to go through the movie sets one-by-one to assign the correct art. If that's not possible, then using MSIF is not much use to me other than providing a storage place for movie sets, to which the user had no access previously, at least to my knowledge. Not an insignificant feature, and I'm glad to have it. I just thought I would be able to replace the KODI-assigned art with my previous assignments. Apparently, that is not the case.
 
If I can delete the libraries and add the content in some other way so that it would grab the set artwork from the MSIF, it would probably be faster than manually assigning set artwork.
 
From: 78andyp ***@***.***> 
Sent: Saturday, 25 November, 2023 7:25 PM
To: xbmc/xbmc ***@***.***>
Cc: hurricane51 ***@***.***>; Comment ***@***.***>
Subject: Re: [xbmc/xbmc] Update blank movie sets with artwork from Movie Set Information Folder… (PR #24131)
 
So it will only populate BLANK movie sets? I did a clean re-install of KODI 19, but my media all had set information in their NFO files. The sets were thus recreated. but KODI generated the set art. I already have a fully populated MSIF folder. How do I reassign the art in my MSIF folder to the existing sets?
Can I just check a couple of things:
You have the MSIF set in Settings -> Media -> Videos -> Movie Set Information Folder??
You have Local Information Only set in the Choose information provider option of the Sources??
If both of those are true then it feels like Kodi should look in the MSIF. I'll see what I can recreate here.
—
Reply to this email directly, view it on GitHub <https://github.com/xbmc/xbmc/pull/24131#issuecomment-1826296224> , or unsubscribe <https://github.com/notifications/unsubscribe-auth/AETWG32HY3BMQEBYF4J2QVDYGHPQFAVCNFSM6AAAAAA7W2YJZKVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMYTQMRWGI4TMMRSGQ> .
You are receiving this because you commented.Message ID: ***@***.***>

ksooo commented:
> I'd actually started playing with a context menu option.

I'm not sure we want/need this, tbh. For all other media types we are managing this using the video info dialog's "refresh" button. I don't see why we should do this differently for movie sets or why we need another way in addition to the dialog.

78andyp commented:
> > I'd actually started playing with a context menu option.
> 
> I'm not sure we want/need this, tbh. AFAIK. for all other media types we are managing this using the video info dialog's "refresh" button. I don't see why we should do this differently for movie sets or why we need another way in addition to the dialog.

Because 'Choose art' option is there already so it makes sense (to me at least) to have the 'Use local art' option there as well.

ksooo commented:
>> Because 'Choose art' option is there already so it makes sense (to me at least) to have the 'Use local art' option there as well

 I see it more the other way round. Having „choose art“ in the context menu is superfluous and should be removed, because we have the dialog for that. Why add same functionality at different places? All this code needs to be maintained…

78andyp commented:
> Yes to both of those. I have been having an exchange on the KODI forum, and here’s what I posted: **************************************** I did a new install of KODI 19. I configured everything before adding any content, and that included specifying the MSIF folder. The MSIF had been created with my previous installation, and I saw no need to re-create it for the same content. My movies all had NFO files along with the video in their individual folder. These NFO's all had set information. I wasn't creating any new sets. I initially specified Use Local Information as my information provider to grab the info from the local files. After the movies were in the library, I changed the scraper to Universal Movie scraper for future additions. I found that KODI had generated/assigned random art (from the included movies) to most of the existing sets. Others, it left blank. It didn't use the MSIF folder art. I would like to avoid having to go through the movie sets one-by-one to assign the correct art. If that's not possible, then using MSIF is not much use to me other than providing a storage place for movie sets, to which the user had no access previously, at least to my knowledge. Not an insignificant feature, and I'm glad to have it. I just thought I would be able to replace the KODI-assigned art with my previous assignments. Apparently, that is not the case. If I can delete the libraries and add the content in some other way so that it would grab the set artwork from the MSIF, it would probably be faster than manually assigning set artwork. From: 78andyp ***@***.***> Sent: Saturday, 25 November, 2023 7:25 PM To: xbmc/xbmc ***@***.***> Cc: hurricane51 ***@***.***>; Comment ***@***.***> Subject: Re: [xbmc/xbmc] Update blank movie sets with artwork from Movie Set Information Folder… (PR #24131) So it will only populate BLANK movie sets? I did a clean re-install of KODI 19, but my media all had set information in their NFO files. The sets were thus recreated. but KODI generated the set art. I already have a fully populated MSIF folder. How do I reassign the art in my MSIF folder to the existing sets? Can I just check a couple of things: You have the MSIF set in Settings -> Media -> Videos -> Movie Set Information Folder?? You have Local Information Only set in the Choose information provider option of the Sources?? If both of those are true then it feels like Kodi should look in the MSIF. I'll see what I can recreate here. — Reply to this email directly, view it on GitHub <[#24131 (comment)](https://github.com/xbmc/xbmc/pull/24131#issuecomment-1826296224)> , or unsubscribe <https://github.com/notifications/unsubscribe-auth/AETWG32HY3BMQEBYF4J2QVDYGHPQFAVCNFSM6AAAAAA7W2YJZKVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMYTQMRWGI4TMMRSGQ> . You are receiving this because you commented.Message ID: ***@***.***>

Have you tried it with the latest beta of Kodi 21?? What OS are you using??

I've just deleted local database completely and manually generated 3 NFOs for Predator, Predator 2 and Predators - naming the Set in the NFO as Predator Collection. Each directory is just the movie MKV and movie.nfo.

I then set the source to local, set single movie in each directory, set scraper to local.

I then set MSIF directory and created subdirectory Predator Collection with just poster.jpg in it.

I then ran scraper for first time and it found the set artwork automatically.

KarellenX commented:
@hurricane51 

The conversations in this PR is strictly for feedback on a new feature, NOT your support requests. Please stop hijacking this conversation.

@78andyp 

Please ignore hurricane51. He was being offered support on the forum, but he discontinued... https://forum.kodi.tv/showthread.php?tid=375147
He does not even have this PR installed and is seeking general help.

78andyp commented:
> > > Because 'Choose art' option is there already so it makes sense (to me at least) to have the 'Use local art' option there as well
> 
> I see it more the other way round. Having „choose art“ in the context menu is superfluous and should be removed, because we have the dialog for that. Why add same functionality at different places? All this code needs to be maintained…

Agree, should probably be both or neither. I personally tend to use context menus (probably as come from a windows centric background).

78andyp commented:
> > Do mean like as a context menu option for the individual set??
> 
> Actually, like ksooo mentions, the Refresh button. Here is an example i've photoshopped... My concern would be for users with large libraries having to run an extensive global Library Update to update one or two movie sets.
> 
> ![setsrefresh](https://private-user-images.githubusercontent.com/44626605/285531066-18a8b62c-b275-41eb-96b7-0fdcfd7bca59.jpg?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTEiLCJleHAiOjE3MDEwMDI1NzksIm5iZiI6MTcwMTAwMjI3OSwicGF0aCI6Ii80NDYyNjYwNS8yODU1MzEwNjYtMThhOGI2MmMtYjI3NS00MWViLTk2YjctMGZkY2ZkN2JjYTU5LmpwZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFJV05KWUFYNENTVkVINTNBJTJGMjAyMzExMjYlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjMxMTI2VDEyMzc1OVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTNlNjdmY2ZiNzU2Nzc5ZjI5MWE3ZWY5YTk4ZWI2MmU4ODNlYjIxMzA5NTJmNWQ3Nzc0ZjQwZjZmZGExOWRlOGEmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.1BJIgA1PovBfb8muLmYHx-pmc8qPsclW8l8eNX54QEA)

I've used the existing refresh button - it now should appear if there it's a blank set and a folder in the MSIF.
I also noticed that changing the art using the Choose Art button doesn't update the library thumbnail when you exit the information dialog (you have to exit the library and go back in) - so I've added a fix for that as well.

78andyp commented on xbmc/video/VideoInfoScanner.cpp:
```diff
@@ -1672,7 +1673,7 @@ namespace VIDEO
     CFileItemList availableArtFiles;
     CDirectory::GetDirectory(path, availableArtFiles,
         CServiceBroker::GetFileExtensionProvider().GetPictureExtensions(),
-        DIR_FLAG_NO_FILE_DIRS | DIR_FLAG_READ_CACHE | DIR_FLAG_NO_FILE_INFO);
```
I changed this as I found that if you add a file to the MSIF once Kodi has started it never gets found on subsequent searches. GIven the MSIF will only contain a small number of files I would have this doesn't have a performance impact.

78andyp commented:
> Have a requirements issue with the use of "refresh". In every other context "refresh" finds a local nfo or configured scraper to update the info. Here, "refresh" won't update the set overview or set membership making it a one-off.

OK, thanks. I'll add an update button.

ksooo commented:
Yet another button in the fossil that is already crowded with (way to much) buttons? And will users know the functional difference between update and refresh. It’s just different words for same thing basically. For example, in German both words would even be the same.

ksooo commented:
If this „refresh“ introduced with this PR it’s functional different from the usual „refresh“ functionality it should have it’s own button, but it’s should have a label that is not generic, but hitting the point.

CrystalP commented on xbmc/video/VideoInfoScanner.cpp:
```diff
@@ -320,6 +320,22 @@ namespace VIDEO
+            {
+              CVideoInfoTag setDetails;
+              m_database.GetSetInfo(movieDetails.m_set.id, setDetails);
+              UpdateSetFromLocalArtwork(setDetails);
```
I'm not sure that's the correct place to trigger the update during library update
1/ Why only for the case of a known dir with no change?
2/ Would be more direct (and avoid a couple db queries) to add a dedicated loop over all sets defined in the db - probably in `CVideoInfoScanner::Process()` as additional step?

78andyp replied:
I've moved it's location in `CVideoInfoScanner::DoScan()` - it seems to fit there as the checking the directory contains `MOVIES`, `.nomedia` screening, checking the directory exists etc.. is done there and would need to be replicated in a seperate `CVideoInfoScanner::Proecss()` loop.

CrystalP commented on xbmc/video/VideoInfoScanner.cpp:
```diff
@@ -435,6 +451,42 @@ namespace VIDEO
+  {
+    bool result = false, isArt = false;
+    m_database.Open();
+    std::map<std::string, std::string> movieArt, setArt;
```
A single declaration per line please, per https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md#52-multiple-declarations

CrystalP commented on xbmc/video/VideoInfoScanner.cpp:
```diff
@@ -435,6 +451,42 @@ namespace VIDEO
+    bool result = false, isArt = false;
+    m_database.Open();
+    std::map<std::string, std::string> movieArt, setArt;
+    isArt = m_database.GetArtForItem(setDetails.m_iDbId, MediaTypeVideoCollection, setArt);
```
const bool isArt = m_database...

CrystalP commented:
First of all thanks for your enthusiasm and many PR! It's just bad timing for reviews with the video versions things and (hopefully) close to a release.

Reading the previous comments, I'd also prefer a "Refresh" button in the info screen, consistent in placement and behavior (in spirit at least) with the other uses of the dialog.

To me it means that the art would be set from the MSIF when blank, and updated from the MSIF when not blank.
A user can change his mind about the art in the MSIF for a set, in the same way that he can change the art in a movie folder and use Refresh to grab the new art,

The processing triggered by library scan (all library or single path) doesn't have to change because of the above, setting only the missing art would be consistent with the way movies are scraped as far as I know.

78andyp commented:
> First of all thanks for your enthusiasm and many PR! It's just bad timing for reviews with the video versions things and (hopefully) close to a release.
> 
> Reading the previous comments, I'd also prefer a "Refresh" button in the info screen, consistent in placement and behavior (in spirit at least) with the other uses of the dialog.

I guess the only difference would be this is not a true refresh - we're not going back to scraper source and updating all the set info, just the art. Also, the languate 'use local art' is used on the context menu - so there would be a discrepancy there.

> 
> To me it means that the art would be set from the MSIF when blank, and updated from the MSIF when not blank. A user can change his mind about the art in the MSIF for a set, in the same way that he can change the art in a movie folder and use Refresh to grab the new art,
>

The code currently doesn't change the art if there is art already present, only if the set is blank.

The wiki says 'If you do not use local artwork, and rely on the scrapers to provide your artwork, then there is no requirement for you to use this feature.' - so this would seem to imply that scrapers should take priority over local art (unless in this case there is no scraped art).
 
> The processing triggered by library scan (all library or single path) doesn't have to change because of the above, setting only the missing art would be consistent with the way movies are scraped as far as I know.

CrystalP commented:
ah yes you're right, a true refresh would attempt to grab the online source first.

CrystalP commented:
ok forget it, maybe as a phase 2 if you feel up to it in the future :-)

78andyp commented:
As far as I can see there are really only 3 elements to a set - the title of the set, the overview of the set and the artwork of the set.
I guess the first two are only likely to change if an additional movie is added, and I assume then that it will be updated if/when that movie is added.

You are right, remote artwork could be added in the future even without a movie content change and a proper refresh should take that into account, but as you say, it would need a popup as it seems local should always trump remote and if this PR sets local artwork now - does it matter if remote artwork is updated in the future.

Happy to look at that for a future PR.

78andyp committed:
[VideoInfoScanner][GUIDialogVideoInfo] Update blank movie sets with artwork from Movie Set Information Folder.

CrystalP commented:
Thinking out loud here, now that I am more familiar with artwork handling, instead of adding a context menu button / button in info dialog, maybe the function should be integrated in the existing Choose art dialog?
By adding a shortcut there to the MSIF of that movie set. Would be available only for movie sets and when there is a subfolder for the movie set being edited:

![image](https://github.com/xbmc/xbmc/assets/489377/ff32fa62-a735-49de-9762-964e39f4d410)

The shortcut would lead to the MSIF of the set, not the top level MSIF folder.
The shortcut could alternatively be in the "Browse..." button, but that would be much less discoverable.

The [example screenshot in the wiki ](https://kodi.wiki/view/Movie_set_information_folder) shows multiple instances of each art type, I don't know how representative of the reality that is - and is your code prepared for that situation?

CrystalP commented on xbmc/video/VideoInfoScanner.cpp:
```diff
@@ -437,6 +452,44 @@ namespace VIDEO
+      m_database.SetArtForItem(setDetails.m_iDbId, MediaTypeVideoCollection, movieSetArt);
+    }
+    m_database.Close();
+    return movieSetArt;
```
It doesn't look like the return value is actually used by the callers and a true/false boolean would be enough?

CrystalP commented on xbmc/video/VideoInfoScanner.cpp:
```diff
@@ -437,6 +452,44 @@ namespace VIDEO
+        m_database.GetArtForItem(setDetails.m_iDbId, MediaTypeVideoCollection, setArt);
+    if (!isArt || !setArt.count("poster"))
+    {
+      // Movie set currently has no artwork poster (other art types do not seem to be used)
```
Estuary uses the fanart when defined in addition to poster. I don't know about the other skins or about the automatic download.

78andyp commented:
> Thinking out loud here, not requesting changes but looking for feedback on the idea. Now that I am more familiar with artwork handling, I'm thinking that instead of adding a context menu button / button in info dialog, maybe the function could be integrated in the existing Choose art dialog (after the selection of the art type)? By adding a shortcut there to the MSIF of that movie set. Would be available only for movie sets and when there is a subfolder for the movie set being edited:
> 
> ![image](https://private-user-images.githubusercontent.com/489377/300648118-ff32fa62-a735-49de-9762-964e39f4d410.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDY3MTkwNzcsIm5iZiI6MTcwNjcxODc3NywicGF0aCI6Ii80ODkzNzcvMzAwNjQ4MTE4LWZmMzJmYTYyLWE3MzUtNDlkZS05NzYyLTk2NGUzOWY0ZDQxMC5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwMTMxJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDEzMVQxNjMyNTdaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT0wODBlNzYzOGQ0MGU0M2FjY2I4N2ZmMWJlMmZlNGY4ZDIyYmJmMjZkNjA2NjQ5ODI2ODVlNGM5MTAwOWIzMTNmJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.eo7TV3DOwZ4n6gCy0DPpuhLUx4EnMfCSrsxJtXAc630)
> 
> The shortcut would lead to the MSIF of the set, not the top level MSIF folder. The shortcut could alternatively be in the "Browse..." button, but that would be much less discoverable.
> 
> The [example screenshot in the wiki ](https://kodi.wiki/view/Movie_set_information_folder) shows multiple instances of each art type, I don't know how representative of the reality that is - and is your code prepared for that situation?

Hi

Thanks for the thoughts.

I was thinking of going down a different route - but I think we can combine the two.

Now we have potential refresh functionality I'm thinking that you anbd @ksoo were right at the start - refresh is better than a separate art update (which I agree with you - would probably be better in the choose art dialog with MSIF added).

I was then going to extend the refresh set PR to make sure when a new set is added the MSIF is checked for art as well. I was also going to add support for a `set.nfo` that could contain an `overview` tag to set the overview (so that it can be changed independent of a `movie.nfo` which may screw up the movie metadata).

Looking to the future I was wondering if the whole set/movie version/(cinematic universe) could be simplified to nested sets. Each (nested) set could contain either further nested sets or streams (linking back to our previous discussions about playlists - a stream would either be 1 stream in a file or a playlist in an ISO). In theory seasons and episodes could be viewed as sets as well (where the overarching term set is really just referring to a collection), as could music videos.

In essence then there would only be 2 significant entities - a stream or a set - at a database level. A set could be designated to be a season, a universe, a version collection, or just a set of sets  (or whatever else we want) and then there would be GUI 'page' assocaited with each set type.

Most set management would still be handled by the scrapers but the MSIF could then become just a set information folder where, for example, you could have within `set.nfo` XML for the cinematic universe order, or the start trek tv series order etc..

KarellenX commented:
Thought I might ping @rmrector as he implemented the MSIF and he had ideas at the time for expanding its functionality.

78andyp commented:
Superceded by #24557.
